src/shared/components/query/QueryStore.ts

```diff
@@ -1684,3 +1658,48 @@ export const QueryStoreComponent = ComponentGetsStoreContext(QueryStore);
 
 const selectedGeneSets = '';
 export default selectedGeneSets;
+
+export function buildTableData(hierarchyData: any[]): Geneset[] {
+	let tableData: Geneset[] = [];
+	for (const node of hierarchyData) {
+	    if (_.has(node, 'genesets')) {
+	        for (const geneset of node.genesets) {
+	            tableData.push({
+	                genesetId: geneset.genesetId,
+	                name: geneset.genesetId,
+	                description : geneset.description,
+	                representativeScore : geneset.representativeScore,
+	                representativePvalue : geneset.representativePvalue,
+	                refLink : geneset.refLink,
+	            });
+	        }
+	    }
+	}
+	return tableData;
+};
+
+export function generateMaxYValue(tableData: Geneset[]|undefined): number|undefined
+{
+    if (tableData) {

```

Comment:
why would tabledata be undefined?  getter to handle this outside of the function.  this function should assume data.